tests/executor/exchange_seq_scan_test.cpp

```diff
@@ -0,0 +1,346 @@
+  std::unique_ptr<storage::DataTable> table(
+      ExecutorTestsUtil::CreateTable(row_num, false));
+  TestingHarness::GetInstance().GetNextTileGroupId();
+  size_t index = 0;
+  ExecutorTestsUtil::PopulateTiles(table->GetTileGroup(index++), row_num);
+  for (size_t i = 0; i < tile_group_num - 1; ++i) {
+    std::vector<catalog::Schema> schemas1(
+        {catalog::Schema({ExecutorTestsUtil::GetColumnInfo(0),
+                          ExecutorTestsUtil::GetColumnInfo(1)}),
+         catalog::Schema({ExecutorTestsUtil::GetColumnInfo(2),
+                          ExecutorTestsUtil::GetColumnInfo(3)})});
+    std::map<oid_t, std::pair<oid_t, oid_t>> column_map1;
+    column_map1[0] = std::make_pair(0, 0);
+    column_map1[1] = std::make_pair(0, 1);
+    column_map1[2] = std::make_pair(1, 0);
+    column_map1[3] = std::make_pair(1, 1);
+    // Create tile groups.
+    table->AddTileGroup(std::shared_ptr<storage::TileGroup>(
+        storage::TileGroupFactory::GetTileGroup(
+            INVALID_OID, INVALID_OID,
+            TestingHarness::GetInstance().GetNextTileGroupId(), table.get(),
+            schemas1, column_map1, row_num)));
+    ExecutorTestsUtil::PopulateTiles(table->GetTileGroup(index++), row_num);
+  }
+  return table.release();
+}
+
+expression::AbstractExpression *CreatePredicate(
+    const std::set<oid_t> &tuple_ids) {
+  assert(tuple_ids.size() >= 1);
+
+  expression::AbstractExpression *predicate =
+      expression::ExpressionUtil::ConstantValueFactory(Value::GetFalse());
+
+  bool even = false;
+  for (oid_t tuple_id : tuple_ids) {
+    even = !even;
+
+    // Create equality expression comparison tuple value and constant value.
+    // First, create tuple value expression.
+    expression::AbstractExpression *tuple_value_expr = nullptr;
+
+    tuple_value_expr =
+        even ? expression::ExpressionUtil::TupleValueFactory(0, 0)
+             : expression::ExpressionUtil::TupleValueFactory(0, 3);
+
+    // Second, create constant value expression.
+    Value constant_value =
+        even ? ValueFactory::GetIntegerValue(
+                   ExecutorTestsUtil::PopulatedValue(tuple_id, 0))
+             : ValueFactory::GetStringValue(std::to_string(
+                   ExecutorTestsUtil::PopulatedValue(tuple_id, 3)));
+
+    expression::AbstractExpression *constant_value_expr =
+        expression::ExpressionUtil::ConstantValueFactory(constant_value);
+
+    // Finally, link them together using an equality expression.
+    expression::AbstractExpression *equality_expr =
+        expression::ExpressionUtil::ComparisonFactory(
+            EXPRESSION_TYPE_COMPARE_EQUAL, tuple_value_expr,
+            constant_value_expr);
+
+    // Join equality expression to other equality expression using ORs.
+    predicate = expression::ExpressionUtil::ConjunctionFactory(
+        EXPRESSION_TYPE_CONJUNCTION_OR, predicate, equality_expr);
+  }
+
+  return predicate;
+}
+
+double GetRunTime(executor::ExchangeSeqScanExecutor &executor,
```

Comment:
These two GetRunTime methods do the same thing and could be merged into one function which just takes an executor since there is no sequential scan specific code.